File: java/MASTER_SLAVE_WHOLE_PART_PROXY/GestorReservacion.java
public class GestorReservacion {
    //Atributos
    private Reservacion reservacionSala = new Reservacion();
    DatosCliente datosCliente; 
    DatosCompra datosCompra;
    DatosFuncion datosFuncion;

    //Constructor 
    public GestorReservacion(){
        //invocaSubordinados(nombre, apellidoP, apellidoM, edad, tipoC, numTarjeta, diaC, mesC, anioC, asiento, dia, mes, anio, pelicula);
    }

    //Métodos
    public void invocaSubordinados(String nombre, String apellidoP, String apellidoM, int edad, String tipoC, String numTarjeta, int diaC, int mesC, int anioC, String asiento, int dia, int mes, int anio, String pelicula){
        datosCliente = GestorDatosCliente.preparaDatosCliente(nombre, apellidoP, apellidoM, edad, tipoC); 
        datosCompra = GestorDatosCompra.preparaDatosCompra(numTarjeta, diaC, mesC, anioC);
        datosFuncion = GestorDatosFuncion.preparaDatosFuncion(asiento, dia, mes, anio, pelicula);
        
        creaReservacion();

        ProxyAutorizacionPago pago = new ProxyAutorizacionPago();
        pago.preparaCargo(datosFuncion);
        boolean resultado = pago.verificacionCargo();
        if(resultado == true){
            System.out.println("RESERVACION EXITOSA");
        }
        else{
            System.out.println("LA RESERVACION FALLO");
        }
    }
    
    public Reservacion unirResultados(){
        return reservacionSala;
    }

    public void creaReservacion(){
        reservacionSala.setDatosCliente(datosCliente);
        reservacionSala.setDatosCompra(datosCompra);
        reservacionSala.setDatosFuncion(datosFuncion);

    }
}

File: php/MASTER_SLAVE_WHOLE_PART_PROXY/principal/GestorReservacion.php
<?php
require './principal/Reservacion.php';
require './gestores/GestorDatosCliente.php';
require './gestores/GestorDatosCompra.php';
require './gestores/GestorDatosFuncion.php';
require './pago/ProxyAutorizacionPago.php';
class GestorReservacion{
    //attr
    private $reservacionSala;
    private $datosCliente;
    private $datosCompra;
    private $datosFuncion;
    
    public function __construct(){
        $this->reservacionSala = new Reservacion(); 

    }
    
    public function invocaSubordinados($nombre, $apellidoP, $apellidoM, $edad, $tipoC, $numTarjeta,
                                        $fecha, $asiento, $dia, $mes, $anio, $pelicula){
    
        $this->datosCliente = GestorDatosCliente::preparaDatosCliente($nombre, $apellidoP, $apellidoM, $edad, $tipoC);
        $this->datosCompra = GestorDatosCompra::preparaDatosCompra($numTarjeta, $fecha);
        $this->datosFuncion = GestorDatosFuncion::preparaDatosFuncion($asiento, $dia, $mes, $anio, $pelicula);
        $this->creaReservacion();
        
        $proxy = new ProxyAutorizacionPago();
        $cargo = new Cargo();

        $r = $proxy->solicitaAutorizacionPago($cargo);
        
        
    }
    
    public function creaReservacion(){
        $this->reservacionSala->setDatosCliente($this->datosCliente);
        $this->reservacionSala->setDatosCompra($this->datosCompra);
        $this->reservacionSala->setDatosFuncion($this->datosFuncion);

    }
    public function unirResultados() {
        return $this->reservacionSala;
    }


    
}



